pat/pat.go

```diff
@@ -132,13 +133,22 @@ const bc = "/.;,"
 var patternRe = regexp.MustCompile(`[` + bc + `]:([^` + bc + `]+)`)
 
 /*
-New returns a new Pattern from the given Pat route. See the package
-documentation for more information about what syntax is accepted by this
-function.
+New returns a new Pattern from the given Pat route and list of
+accepted HTTP methods. If no methods are provided, all methods match. See
+the package documentation for more information about what syntax is
+accepted by this function.
 */
-func New(pat string) *Pattern {
+func New(pat string, methods ...string) *Pattern {
```

Comment:
Agreed, I think it might be nice to provide a `NewWithMethods` as part of the public interface eventually but for now I left it private. Better to wait until someone asks for it and can justify it...